app/src/main/java/org/thoughtcrime/securesms/components/ProfilePictureView.kt

```diff
@@ -3,164 +3,153 @@ package org.thoughtcrime.securesms.components
-        .asDrawable(context, ContactColors.UNKNOWN_COLOR.toConversationColor(context), false, resourcePadding) }
-    private val unknownOpenGroupDrawable by lazy { ResourceContactPhoto(R.drawable.ic_notification)
-        .asDrawable(context, ContactColors.UNKNOWN_COLOR.toConversationColor(context), false, resourcePadding) }
 
-    constructor(context: Context, sender: Recipient): this(context) {
-        update(sender)
+    private val unknownOpenGroupDrawable by lazy(LazyThreadSafetyMode.NONE) {
+        ResourceContactPhoto(R.drawable.ic_notification)
+            .asDrawable(context, ContactColors.UNKNOWN_COLOR.toConversationColor(context), false)
     }
 
-    fun update(recipient: Recipient) {
-        recipient.run { update(address, isClosedGroupRecipient, isOpenGroupInboxRecipient) }
+    private fun setShowAsDoubleMode(showAsDouble: Boolean) {
+        binding.doubleModeImageViewContainer.isVisible = showAsDouble
+        binding.singleModeImageView.isVisible = !showAsDouble
     }
 
-    fun update(
-        address: Address,
-        isClosedGroupRecipient: Boolean = false,
-        isOpenGroupInboxRecipient: Boolean = false
-    ) {
-        fun getUserDisplayName(publicKey: String): String = prefs.takeIf { userPublicKey == publicKey }?.getProfileName()
-            ?: DatabaseComponent.get(context).sessionContactDatabase().getContactWithAccountID(publicKey)?.displayName(Contact.ContactContext.REGULAR)
-            ?: publicKey
-
-        if (isClosedGroupRecipient) {
-            val members = DatabaseComponent.get(context).groupDatabase()
-                .getGroupMemberAddresses(address.toGroupString(), true)
-                .sorted()
-                .take(2)
-            if (members.size <= 1) {
-                publicKey = ""
-                displayName = ""
-                additionalPublicKey = ""
-                additionalDisplayName = ""
-            } else {
-                val pk = members.getOrNull(0)?.serialize() ?: ""
-                publicKey = pk
-                displayName = getUserDisplayName(pk)
-                val apk = members.getOrNull(1)?.serialize() ?: ""
-                additionalPublicKey = apk
-                additionalDisplayName = getUserDisplayName(apk)
+    private fun cancelLastLoadJob() {
+        lastLoadJob?.cancel()
+        lastLoadJob = null
+    }
+
+    @OptIn(DelicateCoroutinesApi::class)
+    private fun loadAsDoubleImages(groupAddress: Address, knownRecipient: Recipient?, allowMemoryCache: Boolean) {
+        cancelLastLoadJob()
+
+        // The use of GlobalScope is intentional here, as there is no better lifecycle scope that we can use
+        // to launch a coroutine from a view. The potential memory leak is not a concern here, as
+        // the coroutine is very short-lived. If you change the code here to be long live then you'll
+        // need to find a better scope to launch the coroutine from.
+        lastLoadJob = GlobalScope.launch(Dispatchers.Main) {
```

Comment:
This is definitely not ideal... The whole view is problematic, which is unfortunately not a new problem. We really shouldn't be having so much heavy logic in a piece of UI...
There's isn't too much else we can do with the current set up... 
This will be part of the things we can fix once we move away from the xml views.